de-lib/src/main/java/org/iplantc/de/client/services/impl/SearchServiceFacadeImpl.java

```diff
@@ -324,67 +270,22 @@ public void saveQueryTemplates(List<DiskResourceQueryTemplate> queryTemplates, A
-        DataSearchQueryBuilder builder = new DataSearchQueryBuilder(queryTemplate, userInfo);
-        String queryParameter = "";
-        String tags = "";
-        
-        String buildFullQuery = builder.buildFullQuery();
-        if (!Strings.isNullOrEmpty(buildFullQuery)) {
-            queryParameter = "q=" + URL.encodeQueryString(buildFullQuery);
-        }
-        String limitParameter = "&limit=" + loadConfig.getLimit();
-        String offsetParameter = "&offset=" + loadConfig.getOffset();
-        String typeParameter = "&type=" + ((searchType == null) ? TYPE.ANY.toString() : searchType.toString());
-        if (!Strings.isNullOrEmpty(builder.taggedWith())) {
-            tags = "&tags=" + URL.encodeQueryString(builder.taggedWith());
-        }
-        String sortParameter = "";
-        List<SortInfoBean> sortInfoList = loadConfig.getSortInfo();
-        if (sortInfoList != null && !sortInfoList.isEmpty()) {
-            SortInfoBean sortInfo = sortInfoList.get(0);
-            String sortField = convertSortField(sortInfo.getSortField());
-            if (!Strings.isNullOrEmpty(sortField)) {
-                String sortDir = sortInfo.getSortDir() == null ? SortDir.ASC.toString() : sortInfo.getSortDir().toString();
-                sortParameter = Format.substitute("&sort={0}:{1}", sortField, sortDir.toLowerCase());
-            }
-        }
+    public void submitSearchQuery(DiskResourceQueryTemplate template,
+                                  FolderContentsLoadConfig loadConfig,
+                                  FolderContentsRpcProxyImpl.SearchResultsCallback queryResultsCallback) {
+        DataSearchQueryBuilder builder = new DataSearchQueryBuilder(template);
+        String address = SEARCH;
 
-        StringBuilder addressSb = new StringBuilder().append(deProperties.getDataMgmtBaseUrl() + "index?");
-        if (!Strings.isNullOrEmpty(queryParameter)) {
-            addressSb.append(queryParameter);
-        }
-
-        if (!Strings.isNullOrEmpty(tags)) {
-            addressSb.append(tags);
-        }
+        Splittable query = builder.getFullQuery();
 
+        int limit = loadConfig.getLimit();
+        int offset = loadConfig.getOffset();
 
-        addressSb.append("&includeTrash="
-                         + queryTemplate.isIncludeTrashItems());
-        addressSb.append(limitParameter);
-        addressSb.append(offsetParameter);
-        addressSb.append(typeParameter);
-        addressSb.append(sortParameter);
-        ServiceCallWrapper wrapper = new ServiceCallWrapper(GET, addressSb.toString());
-        deServiceFacade.getServiceData(wrapper, new SubmitSearchCallbackConverter(callback, queryTemplate, userInfo, drFactory));
+        StringQuoter.create(limit).assign(query, "size");
+        StringQuoter.create(offset).assign(query, "from");
```

Comment:
This is *probably* just me not knowing how StringQuoter and splittables work, but just to verify, this won't produce a string in the output, hopefully? (i.e. it should produce `{"size": 10}` and not `{"size": "10"}`, since the latter won't currently work. probably could make it work if it's a pain, but let me know anyway)

Reply:
Correct - right now it's sending `{"size": 500}`

Also, I will double check the `Shared With` differences - good catch!